### PR TITLE
UIOR-1117: add missing label mapping fields

### PR DESCRIPTION
**Pull request by alisher-epam**
## Purpose

[UIOR-1117](https://issues.folio.org/browse/UIOR-1117) - Do not show special fields in "Version history" for PO and POL
## Approach

https://github.com/folio-org/ui-orders/assets/116072773/645cd298-8de7-4ab6-a9a8-9314372a3df5

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by Dmitriy-Litvinenko — APPROVED** [review]

**Review by usavkov-epam — DISMISSED** [review]

File: src/components/POLine/POLineVersionView/DonorsVersionView/DonorsVersionView.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    const changedDonors = versionContext?.changes?.filter((change) => {
+      return change.path.startsWith('donorOrganizationIds');
+    });
+    const donorIds = changedDonors.map(({ values }) => values.filter(Boolean));
+
+    return donorIds.flat();
+  }, [versionContext?.changes]);
+
+  const renderKeyValueComponent = useCallback(({ value, id, asLink = false }) => {
+    const content = asLink ? <TextLink to={getDonorUrl(id)}>{value}</TextLink> : value;
```
Comment (Terala-Priyanka) [inline src/components/POLine/POLineVersionView/DonorsVersionView/DonorsVersionView.js:30]: Do you need to check if the url is returned from getDonorUrl method?

File: src/components/POLine/POLineVersionView/DonorsVersionView/DonorsVersionView.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    const isUpdated = changedDonorIds?.includes(id);
+
+    const displayValue = isUpdated ? <mark>{content}</mark> : content;
```
Comment (Terala-Priyanka) [inline src/components/POLine/POLineVersionView/DonorsVersionView/DonorsVersionView.js:33]: does this mark tag use default css?
How does it look? Could you add a screencast for the same? Is it in alignment with Folio UX system?

Comment (usavkov-epam) [inline src/components/POLine/POLineVersionView/DonorsVersionView/DonorsVersionView.js:33]: there is util in `stripes-acq-component` to prepare formatted adapted for version history usage - https://github.com/folio-org/stripes-acq-components/blob/master/lib/VersionHistory/hooks/useVersionWrappedFormatter/useVersionWrappedFormatter.js#L6

e.g. of usage https://github.com/folio-org/stripes-acq-components/blob/master/lib/FundDistribution/FundDistributionView/FundDistributionView.js#L101-L105

File: translations/ui-orders/en.json [hunk 3]
```diff
@@ -512,6 +512,7 @@
   "poLine.cancellationRestrictionNote": "Cancellation description",
   "poLine.claimingActive": "Claiming active",
   "poLine.claimingInterval": "Claiming interval",
+  "poLine.donorOrganizationIds": "Donor information",
```
Comment (Terala-Priyanka) [inline translations/ui-orders/en.json:515]: here, translation value doen't match the key. Are you sure to set it this way?

**Review by Terala-Priyanka — DISMISSED** [review]
Could you improve the code coverage?

File: src/components/POLine/POLineVersionView/DonorsVersionView/DonorsVersionView.js (hunk 1 above)
Comment (alisher-epam) [reply src/components/POLine/POLineVersionView/DonorsVersionView/DonorsVersionView.js:30]: I not checking. I am passing boolean to check if I need to return TextLink with value or not

File: translations/ui-orders/en.json (hunk 3 above)
Comment (alisher-epam) [reply translations/ui-orders/en.json:515]: Can you elaborate a bit, please?

Comment (NikitaSedyx) [reply translations/ui-orders/en.json:515]: there is no rule that translation key should match translation value, `donorOrganizationIds` is BE field that comes from history and we need a translation for it

File: src/common/constants/constants.js [hunk 4]
```diff
@@ -46,3 +46,7 @@ export const UNOPEN_ORDER_ABANDONED_HOLDINGS_TYPES = {
 
 export const FIELD_ARRAY_ITEM_IDENTIFIER_KEY = '__key__';
 export const OKAPI_TENANT_HEADER = 'X-Okapi-Tenant';
+
+export const HIDDEN_ORDER_LABELS_MAP = {
```
Comment (usavkov-epam) [inline src/common/constants/constants.js:50]: If this constant is used in several places and should be defined here, please specify the relation with the version history. It's not entirely obvious from the name why the fields are hidden

Comment (usavkov-epam) [reply src/common/constants/constants.js:50]: I believe we have to provide field names to hide, not labels

File: src/components/POLine/POLineVersionView/DonorsVersionView/DonorsVersionView.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  const changedDonorIds = useMemo(() => {
+    const changedDonors = versionContext?.changes?.filter(({ path, type }) => {
+      return path.startsWith('donorOrganizationIds') && type === 'create';
+    });
+    const donorIds = changedDonors.map(({ values }) => values.filter(Boolean));
+
+    return donorIds.flat();
+  }, [versionContext?.changes]);
+
+  const renderKeyValueComponent = useCallback(({ value, id, asLink = false }) => {
```
Comment (usavkov-epam) [inline src/components/POLine/POLineVersionView/DonorsVersionView/DonorsVersionView.js:29]: I doubt whether the name of the function corresponds to its contents

File: src/components/POLine/POLineVersionView/DonorsVersionView/DonorsVersionView.js (hunk 2 above)
Comment (Terala-Priyanka) [reply src/components/POLine/POLineVersionView/DonorsVersionView/DonorsVersionView.js:33]: @alisher-epam 
Thank you for adding a screen cast.
Are you sure that this yellow color highlight is in line with Folio UX? If not, please have a check with Kimie.

File: src/components/POLine/POLineVersionView/DonorsVersionView/DonorsVersionView.js (hunk 1 above)
Comment (Terala-Priyanka) [reply src/components/POLine/POLineVersionView/DonorsVersionView/DonorsVersionView.js:30]: Which Boolean are you referring in your comment?

File: src/common/constants/constants.js (hunk 4 above)
Comment (alisher-epam) [reply src/common/constants/constants.js]: updated

File: src/components/POLine/POLineVersionView/DonorsVersionView/DonorsVersionView.js (hunk 5 above)
Comment (alisher-epam) [reply src/components/POLine/POLineVersionView/DonorsVersionView/DonorsVersionView.js]: updated

**Review by usavkov-epam — APPROVED** [review]

**Comment by alisher-epam** [comment]
@NikitaSedyx @Terala-Priyanka Are you ok IF we merge this PR?

**Comment by alisher-epam** [comment]
@Terala-Priyanka we need your attention here.

**Pull request merged**